src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -128,6 +133,112 @@ internal partial class Binder
             { WasCompilerGenerated = wasCompilerGenerated };
         }
 
+        private BoundExpression CreateImplicitNewConversion(SyntaxNode syntax, BoundExpression source, Conversion conversion, bool isCast, TypeSymbol destination, DiagnosticBag diagnostics)
+        {
+            var node = (ObjectCreationExpressionSyntax)source.Syntax;
+
+            var arguments = AnalyzedArguments.GetInstance();
+            BindArgumentsAndNames(node.ArgumentList, diagnostics, arguments, allowArglist: true);
+
+            var strippedType = destination.StrippedType();
+            BoundObjectInitializerExpressionBase boundInitializerOpt = node.Initializer != null
+                ? BindInitializerExpression(syntax: node.Initializer, type: strippedType, typeSyntax: syntax, diagnostics)
+                : null;
+
+            BoundExpression operand = source;
+            bool hasErrors;
+            switch (strippedType.TypeKind)
+            {
+                case TypeKind.Array:
+                case TypeKind.Enum:
+                case TypeKind.Delegate:
+                case TypeKind.Interface:
+                    Error(diagnostics, ErrorCode.ERR_BadTargetTypeForNew, syntax, strippedType);
+                    hasErrors = true;
+                    break;
+
+                case TypeKind.Pointer:
+                    Error(diagnostics, ErrorCode.ERR_UnsafeTypeInObjectCreation, syntax, strippedType);
+                    hasErrors = true;
+                    break;
+
+                case TypeKind.Dynamic:
+                    Error(diagnostics, ErrorCode.ERR_NoConstructors, syntax, strippedType);
+                    hasErrors = true;
+                    break;
+
+                case TypeKind.Struct when strippedType.IsTupleType:
+                    Error(diagnostics, ErrorCode.ERR_NewWithTupleTypeSyntax, syntax, strippedType);
+                    hasErrors = true;
+                    break;
+
+                case TypeKind.Struct:
+                case TypeKind.Class:
+                    operand = BindClassCreationExpression(
+                        node,
+                        typeName: strippedType.Name,
+                        typeNode: node,
+                        type: (NamedTypeSymbol)strippedType,
+                        arguments,
+                        diagnostics,
+                        boundInitializerOpt,
+                        forTargetTypedNew: true);
+                    operand.WasCompilerGenerated = true;
```

Comment:
>WasCompilerGenerated [](start = 28, length = 20)

I don't think this should be considered compiler-generated. It is the only representation in the initial bound tree of the creation expression.

Reply:
Generally, we set the "WasCompilerGenerated" flag when a node is created (so we can pretend it is immutable) rather than mutating a node we get back from binding something.

---
In reply to: [204927207](https://github.com/dotnet/roslyn/pull/25196#discussion_r204927207) [](ancestors = 204927207)

Reply:
>rather than mutating a node we get back from binding something.

I think we do that if we generated the underlying node as well, which I agree it's not the case here.